common/types/replicator.go

```diff
@@ -649,6 +649,9 @@ func (v *ReplicationMessages) GetEarliestCreationTime() *int64 {
 	return &result
 }
 
+// ReplicationMessagesSizeFn is a function type to calculate size of ReplicationMessages
+type ReplicationMessagesSizeFn func(v *ReplicationMessages) int
```

Comment:
@timl3136  it might be worth landing your size work at some point so it can be used here as an interface